### PR TITLE
new play-Download API response

### PR DESCRIPTION
**Pull request by abhi-works**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by abhi-works** [comment]
Screenshot 2022-10-06 at 9 08 18 PM
  Snapshot of Play

**Comment by atapas** [comment]
@abhi-works Have you created an issue for this?

**Comment by abhi-works** [comment]
> @abhi-works Have you created an issue for this?

Yes Tapas https://github.com/reactplay/react-play/issues/629

**Comment by atapas** [comment]
> > @abhi-works Have you created an issue for this?
> 
> Yes @atapas #629 .Anything else pending from my side ?

Nope, started the review. Btw, this is an interesting project. Thanks!

**Comment by atapas** [comment]
@abhi-works One comment on look and feel. I think a lighter background will be better

image

File: src/plays/download-fetch-response/CustomHooks/UseCustomFetchAPI.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from "react";
+
+export const UseCustomFetchAPI = (endPoint) => {
+  const [data, setData] = useState();
+  const [error, setError] = useState(false);
+  const [fetching, setFetching] = useState(true);
+
+  useEffect(() => {
+    const controller = new AbortController();
```
Comment (atapas) [inline src/plays/download-fetch-response/CustomHooks/UseCustomFetchAPI.jsx:9]: A line of comments on `AbortController` will be good.

File: src/plays/download-fetch-response/Readme.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- Language: js
+- Level: Beginner
+
+## Implementation Details
+
+To accomplish this, a custom hook for the fetch (GET)API is written along with UseEffect which includes a cleanup function .
+
+## Consideration
+
+Works for all valid get API's(provided token is not mandatory)
```
Comment (atapas) [inline src/plays/download-fetch-response/Readme.md:16]: I would suggest we add this as a note in the UI too.

File: src/plays/download-fetch-response/DownloadFetchResponse.jsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            <CardActions>
+              <Button
+                disabled={["", null, undefined].includes(endpoint.trim())}
+                onClick={() => setToBeFetched(endpoint)}
+                size="small"
+              >
+                Download
+              </Button>
+            </CardActions>
+          </Card>{" "}
```
Comment (atapas) [inline src/plays/download-fetch-response/DownloadFetchResponse.jsx:81]: Can we also add a `Clear` option to clear the text?

**Review by atapas — CHANGES_REQUESTED** [review]
@abhi-works My reviews are in. All the best.

File: src/plays/download-fetch-response/DownloadFetchResponse.jsx (hunk 3 above)
Comment (abhi-works) [reply src/plays/download-fetch-response/DownloadFetchResponse.jsx:89]: Added the clear option

File: src/plays/download-fetch-response/Readme.md (hunk 2 above)
Comment (abhi-works) [reply src/plays/download-fetch-response/Readme.md:16]: Added

File: src/plays/download-fetch-response/CustomHooks/UseCustomFetchAPI.jsx (hunk 1 above)
Comment (abhi-works) [reply src/plays/download-fetch-response/CustomHooks/UseCustomFetchAPI.jsx:10]: Added

**Comment by abhi-works** [comment]
> @abhi-works My reviews are in. All the best.

Thanks @atapas .Changes are pushed and ready for review

File: src/plays/download-fetch-response/DownloadFetchResponse.jsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  useEffect(() => {
+    setAlertError(error);
+  }, [error]);
+
+  const reset = () => {
+    setEndpoint("");
+    setToBeFetched("");
+  };
+  return (
+    <div className="fetch-download">
```
Comment (atapas) [inline src/plays/download-fetch-response/DownloadFetchResponse.jsx:45]: @abhi-works 

It seems you have removed all the code that the create-play process generates. That results into losing the play headre. You can cross check with other plays. Can you please bring it back?

**Review by atapas — CHANGES_REQUESTED** [review]
This needs to be fixed.

File: src/plays/download-fetch-response/DownloadFetchResponse.jsx (hunk 4 above)
Comment (abhi-works) [reply src/plays/download-fetch-response/DownloadFetchResponse.jsx]: Got It but the template is not generated with cli npx command

Comment (atapas) [reply src/plays/download-fetch-response/DownloadFetchResponse.jsx]: That's a problem that we have to debug then and fix in ReactPlay...

In the mean time, you can understand the structure from other plays and do for yours.

Here is an example: https://github.com/reactplay/react-play/blob/main/src/plays/image-gallery/Imagegallery.js

Also please let us know on Discord, what error you get while creating the play. We want to fix that asap.

Comment (abhi-works) [reply src/plays/download-fetch-response/DownloadFetchResponse.jsx]: Yes, considering other plays, I have included the header for the time being. Changes are pushed . I'll try to make another play and see if the problem persists @atapas

File: src/plays/download-fetch-response/DownloadFetchResponse.jsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+  const reset = () => {
+    setEndpoint("");
+    setToBeFetched("");
+  };
+  return (
+    <div className="play-details">
+      <PlayHeader play={props} />
+      {/* Your Code Starts Here */}
+      <div className="fetch-download">
```
Comment (atapas) [inline src/plays/download-fetch-response/DownloadFetchResponse.jsx:49]: This should be wrapped inside
```html
<div className="play-details-body">
....
</div>
```

**Review by atapas — CHANGES_REQUESTED** [review]
One more change

File: src/plays/download-fetch-response/DownloadFetchResponse.jsx (hunk 5 above)
Comment (abhi-works) [reply src/plays/download-fetch-response/DownloadFetchResponse.jsx]: Pushed requested changes @atapas

**Comment by atapas** [comment]
You have taken care of all my code review comments @abhi-works Looks good... but you need to take care of the responsiveness too... mostly my last review comment on this PR :)

image

**Comment by abhi-works** [comment]
> You have taken care of all my code review comments @abhi-works Looks good... but you need to take care of the responsiveness too... mostly my last review comment on this PR :)
> 
> image

 
Screenshot 2022-10-08 at 7 37 21 PM
Hope this fix works @atapas .Pushed the changes

**Comment by abhi-works** [comment]
Thanks! @atapas .Pushed fix to address the responsive screen  issue with help of material library hook which detects screen size.Hope this media screen fix works

**Review by atapas — APPROVED** [review]
@abhi-works Looks good.

**Pull request merged**